lib/hlclock/timestamp.ex

```diff
@@ -97,6 +130,19 @@ defmodule HLClock.Timestamp do
     %T{time: t, counter: c, node_id: n}
   end
 
+  @doc """
+  Recover a Timestamp from the string representation.
+  """
+  def from_string(tstr) do
+    with {:ok, dt, _} <- tstr |> String.slice(0, 24) |> DateTime.from_iso8601(),
```

Comment:
I think we can replace all the slices with a `[time, counter, node] = String.split("-")`

Reply:
You could with an extra step. Time would be three strings that would have to be rejoined.
`"2017-10-21T17:19:55.859Z-0000-0000000005474CE1" # ["2017", "10", "21T17:19:55.859Z", "0000", "0000000005474CE1"]`. Would it be better, in your opinion, to rejoin the time components this way or do the slice and dice?